### ISSUE TITLE
Regression with Micronaut and Thymeleaf: com.oracle.svm.hosted.substitute.DeletedElementException

### ISSUE THREAD

**Issue by ilopmar** [oracle/graal#3123]
**Describe the issue**

Our CI has detected a regression with Thymeleaf support in Micronaut in GraalVM 21.0 release branch.

When trying to build the native image it fails with `com.oracle.svm.hosted.substitute.DeletedElementException` (see exception below). 

This didn't happen like a month ago (sorry, I don't have the exact commit when this was introduced).

Please note that adding `--report-unsupported-elements-at-runtime` as a workaround makes it work again.

I've also tried with the latest changes from `master` branch and it also fails with the same error.

**Steps to reproduce the issue**

- `git clone https://github.com/micronaut-graal-tests/micronaut-views-graal`
- `cd micronaut-views-graal`
- `git checkout error-thymeleaf`
- Use GraalVM release branch (21.0)
- `./build-native-image.sh`

**Describe GraalVM and your environment:**
 - GraalVM version: **21.0 release branch (ef5905744db11a90c6c15db99efd0ba66cfc3a4d) and master (48eac58356998e6c6450986966ce955b1445039e)**
 - JDK major version: **8 and 11**
 - OS: **Linux Mint 20**
 - Architecture: **AMD64**

**More details**

```
Executing [
/home/ivan/workspaces/misc/graal/graal/sdk/mxbuild/linux-amd64/GRAALVM_E577E0E91A_JAVA8/graalvm-e577e0e91a-java8-21.0.0-dev/jre/bin/java \
-XX:+UseParallelGC \
-XX:+UnlockExperimentalVMOptions \
-XX:+EnableJVMCI \
-Dtruffle.TrustAllTruffleRuntimeProviders=true \
-Dtruffle.TruffleRuntime=com.oracle.truffle.api.impl.DefaultTruffleRuntime \
-Dgraalvm.ForcePolyglotInvalid=true \
-Dgraalvm.locatorDisabled=true \
-d64 \
-XX:-UseJVMCIClassLoader \
-XX:+UseJVMCINativeLibrary \
-Xss10m \
-Xms1g \
-Xmx14g \
-Duser.country=US \
-Duser.language=en \
-Djava.awt.headless=true \
-Dorg.graalvm.version=21.0.0-dev \
-Dorg.graalvm.config= \
-Dcom.oracle.graalvm.isaot=true \
-Djava.system.class.loader=com.oracle.svm.hosted.NativeImageSystemClassLoader \
-Xshare:off \
-Djvmci.class.path.append=/home/ivan/workspaces/misc/graal/graal/sdk/mxbuild/linux-amd64/GRAALVM_E577E0E91A_JAVA8/graalvm-e577e0e91a-java8-21.0.0-dev/jre/lib/jvmci/graal.jar \
-Djdk.internal.lambda.disableEagerInitialization=true \
-Djdk.internal.lambda.eagerlyInitialize=false \
-Djava.lang.invoke.InnerClassLambdaMetafactory.initializeLambdas=false \
-javaagent:/home/ivan/workspaces/misc/graal/graal/sdk/mxbuild/linux-amd64/GRAALVM_E577E0E91A_JAVA8/graalvm-e577e0e91a-java8-21.0.0-dev/jre/lib/svm/builder/svm.jar \
-Xbootclasspath/a:/home/ivan/workspaces/misc/graal/graal/sdk/mxbuild/linux-amd64/GRAALVM_E577E0E91A_JAVA8/graalvm-e577e0e91a-java8-21.0.0-dev/jre/lib/boot/graal-sdk.jar \
-cp \
/home/ivan/workspaces/misc/graal/graal/sdk/mxbuild/linux-amd64/GRAALVM_E577E0E91A_JAVA8/graalvm-e577e0e91a-java8-21.0.0-dev/jre/lib/svm/builder/llvm-wrapper-shadowed.jar:/home/ivan/workspaces/misc/graal/graal/sdk/mxbuild/linux-amd64/GRAALVM_E577E0E91A_JAVA8/graalvm-e577e0e91a-java8-21.0.0-dev/jre/lib/svm/builder/svm.jar:/home/ivan/workspaces/misc/graal/graal/sdk/mxbuild/linux-amd64/GRAALVM_E577E0E91A_JAVA8/graalvm-e577e0e91a-java8-21.0.0-dev/jre/lib/svm/builder/llvm-platform-specific-shadowed.jar:/home/ivan/workspaces/misc/graal/graal/sdk/mxbuild/linux-amd64/GRAALVM_E577E0E91A_JAVA8/graalvm-e577e0e91a-java8-21.0.0-dev/jre/lib/svm/builder/javacpp-shadowed.jar:/home/ivan/workspaces/misc/graal/graal/sdk/mxbuild/linux-amd64/GRAALVM_E577E0E91A_JAVA8/graalvm-e577e0e91a-java8-21.0.0-dev/jre/lib/svm/builder/svm-llvm.jar:/home/ivan/workspaces/misc/graal/graal/sdk/mxbuild/linux-amd64/GRAALVM_E577E0E91A_JAVA8/graalvm-e577e0e91a-java8-21.0.0-dev/jre/lib/svm/builder/pointsto.jar:/home/ivan/workspaces/misc/graal/graal/sdk/mxbuild/linux-amd64/GRAALVM_E577E0E91A_JAVA8/graalvm-e577e0e91a-java8-21.0.0-dev/jre/lib/svm/builder/objectfile.jar:/home/ivan/workspaces/misc/graal/graal/sdk/mxbuild/linux-amd64/GRAALVM_E577E0E91A_JAVA8/graalvm-e577e0e91a-java8-21.0.0-dev/jre/lib/jvmci/graal-management.jar:/home/ivan/workspaces/misc/graal/graal/sdk/mxbuild/linux-amd64/GRAALVM_E577E0E91A_JAVA8/graalvm-e577e0e91a-java8-21.0.0-dev/jre/lib/jvmci/graal.jar:/home/ivan/workspaces/misc/graal/graal/sdk/mxbuild/linux-amd64/GRAALVM_E577E0E91A_JAVA8/graalvm-e577e0e91a-java8-21.0.0-dev/jre/lib/jvmci/jvmci-api.jar:/home/ivan/workspaces/misc/graal/graal/sdk/mxbuild/linux-amd64/GRAALVM_E577E0E91A_JAVA8/graalvm-e577e0e91a-java8-21.0.0-dev/jre/lib/jvmci/jvmci-hotspot.jar:/home/ivan/workspaces/misc/graal/graal/sdk/mxbuild/linux-amd64/GRAALVM_E577E0E91A_JAVA8/graalvm-e577e0e91a-java8-21.0.0-dev/jre/lib/jvmci/graal-truffle-jfr-impl.jar:/home/ivan/workspaces/misc/graal/graal/sdk/mxbuild/linux-amd64/GRAALVM_E577E0E91A_JAVA8/graalvm-e577e0e91a-java8-21.0.0-dev/jre/lib/resources.jar \
com.oracle.svm.hosted.NativeImageGeneratorRunner \
-watchpid \
74601 \
-imagecp \
/home/ivan/workspaces/misc/graal/graal/sdk/mxbuild/linux-amd64/GRAALVM_E577E0E91A_JAVA8/graalvm-e577e0e91a-java8-21.0.0-dev/jre/lib/boot/graal-sdk.jar:/home/ivan/workspaces/misc/graal/graal/sdk/mxbuild/linux-amd64/GRAALVM_E577E0E91A_JAVA8/graalvm-e577e0e91a-java8-21.0.0-dev/jre/lib/svm/library-support.jar:/home/ivan/micronaut-views-graal/build/libs/views-graal-0.1-all.jar \
-H:Path=/home/ivan/micronaut-views-graal \
-H:FallbackThreshold=0 \
-H:ClassInitialization=io.netty.util.AbstractReferenceCounted:run_time,io.netty.util.concurrent.GlobalEventExecutor:run_time,io.netty.util.concurrent.ImmediateEventExecutor:run_time,io.netty.util.concurrent.ScheduledFutureTask:run_time,io.netty.util.internal.ThreadLocalRandom:run_time \
'-H:ClassInitialization=io.netty.util.NetUtilSubstitutions$NetUtilLocalhost4LazyHolder:run_time' \
'-H:ClassInitialization=io.netty.util.NetUtilSubstitutions$NetUtilLocalhost6LazyHolder:run_time' \
'-H:ClassInitialization=io.netty.util.NetUtilSubstitutions$NetUtilLocalhostLazyHolder:run_time' \
-H:ClassInitialization=io.netty.buffer.PooledByteBufAllocator:run_time,io.netty.buffer.ByteBufAllocator:run_time,io.netty.buffer.ByteBufUtil:run_time,io.netty.buffer.AbstractReferenceCountedByteBuf:run_time \
-H:ReflectionConfigurationResources=META-INF/native-image/io.netty/transport/reflection-config.json \
-H:ClassInitialization=io.netty.handler.ssl.util.ThreadLocalInsecureRandom:run_time \
-H:ClassInitialization=io.netty.handler.codec.http.HttpObjectEncoder:run_time,io.netty.handler.codec.http.websocketx.WebSocket00FrameEncoder:run_time,io.netty.handler.codec.http.websocketx.extensions.compression.DeflateDecoder:run_time \
-H:ClassInitialization=io.netty:build_time \
-H:ClassInitialization=io.netty.handler.codec.http2.Http2CodecUtil:run_time,io.netty.handler.codec.http2.Http2ClientUpgradeCodec:run_time,io.netty.handler.codec.http2.Http2ConnectionHandler:run_time,io.netty.handler.codec.http2.DefaultHttp2FrameWriter:run_time \
-H:IncludeResources=META-INF/http/mime.types \
-H:ClassInitialization=io.micronaut.http.bind.binders.ContinuationArgumentBinder:run_time \
-H:ClassInitialization=io.micronaut.views.csp.CspConfiguration:run_time \
-H:ClassInitialization=io.micronaut.core.io.socket.SocketUtils:run_time \
-H:+AllowIncompleteClasspath \
-H:EnableURLProtocols=http,https \
'-H:ClassInitialization=io.netty.handler.ssl.ConscryptAlpnSslEngine:run_time,io.netty.handler.ssl.JettyNpnSslEngine:run_time,io.netty.handler.ssl.ReferenceCountedOpenSslEngine:run_time,io.netty.handler.ssl.JdkNpnApplicationProtocolNegotiator:run_time,io.netty.handler.ssl.ReferenceCountedOpenSslServerContext:run_time,io.netty.handler.ssl.ReferenceCountedOpenSslClientContext:run_time,io.netty.handler.ssl.util.BouncyCastleSelfSignedCertGenerator:run_time,io.netty.handler.ssl.ReferenceCountedOpenSslContext:run_time,io.micronaut.buffer.netty.NettyByteBufferFactory:run_time,io.netty.handler.ssl.JettyAlpnSslEngine$ClientEngine:run_time,io.netty.handler.ssl.JettyAlpnSslEngine$ServerEngine:run_time,io.netty.handler.codec.http2.Http2CodecUtil:run_time,io.netty.handler.codec.http2.CleartextHttp2ServerUpgradeHandler:run_time,io.netty.handler.codec.http2.Http2ServerUpgradeCodec:run_time,io.micronaut.http.netty.channel.converters.EpollChannelOptionFactory:run_time,io.micronaut.http.netty.channel.converters.KQueueChannelOptionFactory:run_time,io.micronaut.http.bind.binders.ContinuationArgumentBinder$Companion:run_time,io.micronaut.http.bind.binders.ContinuationArgumentBinder:run_time' \
-H:+InstallExitHandlers \
-H:ClassInitialization=io.micronaut.reactive.reactor.ReactorInstrumentation:run_time \
-H:ClassInitialization=ch.qos.logback:build_time,com.fasterxml.jackson:build_time,io.micronaut:build_time,io.reactivex:build_time,org.reactivestreams:build_time,org.slf4j:build_time,org.yaml.snakeyaml:build_time,javax.xml:build_time \
-H:ClassInitialization=com.sun.org.apache.xerces.internal.util:build_time,com.sun.org.apache.xerces.internal.impl:build_time,jdk.xml.internal:build_time,com.sun.xml.internal.stream.util:build_time,com.sun.org.apache.xerces.internal.xni:build_time,com.sun.org.apache.xerces.internal.utils:build_time \
-H:ClassInitialization=javassist.ClassPool:build_time \
-H:Name=views-thymeleaf \
-H:Class=example.micronaut.Application \
-H:ReflectionConfigurationResources=META-INF/native-image/io.micronaut/micronaut-core/reflect-config.json \
-H:ReflectionConfigurationResources=META-INF/native-image/io.micronaut/micronaut-http/reflect-config.json \
-H:ReflectionConfigurationResources=META-INF/native-image/io.micronaut/micronaut-http-netty/reflect-config.json \
-H:ReflectionConfigurationResources=META-INF/native-image/io.micronaut/micronaut-runtime/reflect-config.json \
-H:ReflectionConfigurationResources=META-INF/native-image/io.micronaut/micronaut-http-server-netty/reflect-config.json \
-H:ReflectionConfigurationResources=META-INF/native-image/example.micronaut/views-graal/reflect-config.json \
-H:ResourceConfigurationResources=META-INF/native-image/io.micronaut/micronaut-http/resource-config.json \
-H:ResourceConfigurationResources=META-INF/native-image/example.micronaut/views-graal/resource-config.json \
-H:+ReportExceptionStackTraces \
-H:CLibraryPath=/home/ivan/workspaces/misc/graal/graal/sdk/mxbuild/linux-amd64/GRAALVM_E577E0E91A_JAVA8/graalvm-e577e0e91a-java8-21.0.0-dev/jre/lib/svm/clibraries/linux-amd64 \

]
[views-thymeleaf:74625]    classlist:   1,988.00 ms,  2.13 GB
[views-thymeleaf:74625]        (cap):     458.26 ms,  2.13 GB
[views-thymeleaf:74625]        setup:   1,504.85 ms,  2.13 GB
[views-thymeleaf:74625]     (clinit):   1,083.49 ms,  2.94 GB
[views-thymeleaf:74625]   (typeflow):  15,678.49 ms,  2.94 GB
[views-thymeleaf:74625]    (objects):  17,976.44 ms,  2.94 GB
[views-thymeleaf:74625]   (features):   1,458.48 ms,  2.94 GB
[views-thymeleaf:74625]     analysis:  38,151.38 ms,  2.94 GB
Error: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method java.lang.ClassLoader.defineClass(String, byte[], int, int, ProtectionDomain) is reachable
To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
Detailed message:
Trace: 
	at parsing ognl.security.OgnlSecurityManagerFactory.<init>(OgnlSecurityManagerFactory.java:52)
Call path from entry point to ognl.security.OgnlSecurityManagerFactory.<init>(): 
	at ognl.security.OgnlSecurityManagerFactory.<init>(OgnlSecurityManagerFactory.java:44)
	at ognl.security.OgnlSecurityManagerFactory.getOgnlSecurityManager(OgnlSecurityManagerFactory.java:30)
	at ognl.OgnlRuntime.invokeMethodInsideSandbox(OgnlRuntime.java:1243)
	at ognl.OgnlRuntime.invokeMethod(OgnlRuntime.java:1211)
	at ognl.OgnlRuntime.callAppropriateMethod(OgnlRuntime.java:1920)
	at ognl.ObjectMethodAccessor.callMethod(ObjectMethodAccessor.java:68)
	at ognl.OgnlRuntime.callMethod(OgnlRuntime.java:1996)
	at ognl.ASTMethod.getValueBody(ASTMethod.java:91)
	at ognl.SimpleNode.evaluateGetValueBody(SimpleNode.java:212)
	at ognl.SimpleNode.getValue(SimpleNode.java:236)
	at ognl.Ognl.getValue(Ognl.java:537)
	at ognl.Ognl.getValue(Ognl.java:501)
	at org.thymeleaf.standard.expression.OGNLVariableExpressionEvaluator.executeExpression(OGNLVariableExpressionEvaluator.java:328)
	at org.thymeleaf.standard.expression.OGNLVariableExpressionEvaluator.evaluate(OGNLVariableExpressionEvaluator.java:170)
	at org.thymeleaf.standard.expression.OGNLVariableExpressionEvaluator.evaluate(OGNLVariableExpressionEvaluator.java:95)
	at org.thymeleaf.standard.expression.VariableExpression.executeVariableExpression(VariableExpression.java:166)
	at org.thymeleaf.standard.expression.SimpleExpression.executeSimple(SimpleExpression.java:66)
	at org.thymeleaf.standard.expression.Expression.execute(Expression.java:109)
	at org.thymeleaf.standard.expression.Expression.execute(Expression.java:138)
	at org.thymeleaf.standard.processor.StandardDefaultAttributesTagProcessor.processDefaultAttribute(StandardDefaultAttributesTagProcessor.java:165)
	at org.thymeleaf.standard.processor.StandardDefaultAttributesTagProcessor.process(StandardDefaultAttributesTagProcessor.java:98)
	at org.thymeleaf.engine.ProcessorTemplateHandler.handleStandaloneElement(ProcessorTemplateHandler.java:918)
	at org.thymeleaf.engine.StandaloneElementTag.beHandled(StandaloneElementTag.java:228)
	at org.thymeleaf.engine.TemplateModel.process(TemplateModel.java:136)
	at org.thymeleaf.engine.TemplateManager.process(TemplateManager.java:519)
	at org.thymeleaf.util.LazyProcessingCharSequence.resolveText(LazyProcessingCharSequence.java:78)
	at org.thymeleaf.util.AbstractLazyCharSequence.getText(AbstractLazyCharSequence.java:68)
	at org.thymeleaf.util.AbstractLazyCharSequence.hashCode(AbstractLazyCharSequence.java:132)
	at java.util.HashMap.hash(HashMap.java:339)
	at java.util.HashMap.get(HashMap.java:557)
	at com.oracle.svm.core.jdk.Target_java_nio_charset_Charset.lookup(CharsetSubstitutions.java:74)
	at java.nio.charset.Charset.isSupported(Charset.java:505)
	at com.oracle.svm.jni.JNIJavaCallWrappers.jniInvoke_VA_LIST:Ljava_nio_charset_Charset_2_0002eisSupported_00028Ljava_lang_String_2_00029Z(generated:0)

com.oracle.svm.core.util.UserError$UserException: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method java.lang.ClassLoader.defineClass(String, byte[], int, int, ProtectionDomain) is reachable
To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
Detailed message:
Trace: 
	at parsing ognl.security.OgnlSecurityManagerFactory.<init>(OgnlSecurityManagerFactory.java:52)
Call path from entry point to ognl.security.OgnlSecurityManagerFactory.<init>(): 
	at ognl.security.OgnlSecurityManagerFactory.<init>(OgnlSecurityManagerFactory.java:44)
	at ognl.security.OgnlSecurityManagerFactory.getOgnlSecurityManager(OgnlSecurityManagerFactory.java:30)
	at ognl.OgnlRuntime.invokeMethodInsideSandbox(OgnlRuntime.java:1243)
	at ognl.OgnlRuntime.invokeMethod(OgnlRuntime.java:1211)
	at ognl.OgnlRuntime.callAppropriateMethod(OgnlRuntime.java:1920)
	at ognl.ObjectMethodAccessor.callMethod(ObjectMethodAccessor.java:68)
	at ognl.OgnlRuntime.callMethod(OgnlRuntime.java:1996)
	at ognl.ASTMethod.getValueBody(ASTMethod.java:91)
	at ognl.SimpleNode.evaluateGetValueBody(SimpleNode.java:212)
	at ognl.SimpleNode.getValue(SimpleNode.java:236)
	at ognl.Ognl.getValue(Ognl.java:537)
	at ognl.Ognl.getValue(Ognl.java:501)
	at org.thymeleaf.standard.expression.OGNLVariableExpressionEvaluator.executeExpression(OGNLVariableExpressionEvaluator.java:328)
	at org.thymeleaf.standard.expression.OGNLVariableExpressionEvaluator.evaluate(OGNLVariableExpressionEvaluator.java:170)
	at org.thymeleaf.standard.expression.OGNLVariableExpressionEvaluator.evaluate(OGNLVariableExpressionEvaluator.java:95)
	at org.thymeleaf.standard.expression.VariableExpression.executeVariableExpression(VariableExpression.java:166)
	at org.thymeleaf.standard.expression.SimpleExpression.executeSimple(SimpleExpression.java:66)
	at org.thymeleaf.standard.expression.Expression.execute(Expression.java:109)
	at org.thymeleaf.standard.expression.Expression.execute(Expression.java:138)
	at org.thymeleaf.standard.processor.StandardDefaultAttributesTagProcessor.processDefaultAttribute(StandardDefaultAttributesTagProcessor.java:165)
	at org.thymeleaf.standard.processor.StandardDefaultAttributesTagProcessor.process(StandardDefaultAttributesTagProcessor.java:98)
	at org.thymeleaf.engine.ProcessorTemplateHandler.handleStandaloneElement(ProcessorTemplateHandler.java:918)
	at org.thymeleaf.engine.StandaloneElementTag.beHandled(StandaloneElementTag.java:228)
	at org.thymeleaf.engine.TemplateModel.process(TemplateModel.java:136)
	at org.thymeleaf.engine.TemplateManager.process(TemplateManager.java:519)
	at org.thymeleaf.util.LazyProcessingCharSequence.resolveText(LazyProcessingCharSequence.java:78)
	at org.thymeleaf.util.AbstractLazyCharSequence.getText(AbstractLazyCharSequence.java:68)
	at org.thymeleaf.util.AbstractLazyCharSequence.hashCode(AbstractLazyCharSequence.java:132)
	at java.util.HashMap.hash(HashMap.java:339)
	at java.util.HashMap.get(HashMap.java:557)
	at com.oracle.svm.core.jdk.Target_java_nio_charset_Charset.lookup(CharsetSubstitutions.java:74)
	at java.nio.charset.Charset.isSupported(Charset.java:505)
	at com.oracle.svm.jni.JNIJavaCallWrappers.jniInvoke_VA_LIST:Ljava_nio_charset_Charset_2_0002eisSupported_00028Ljava_lang_String_2_00029Z(generated:0)

	at com.oracle.svm.core.util.UserError.abort(UserError.java:82)
	at com.oracle.svm.hosted.FallbackFeature.reportAsFallback(FallbackFeature.java:233)
	at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:773)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:563)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$0(NativeImageGenerator.java:476)
	at java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1386)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:175)
Caused by: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method java.lang.ClassLoader.defineClass(String, byte[], int, int, ProtectionDomain) is reachable
To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
Detailed message:
Trace: 
	at parsing ognl.security.OgnlSecurityManagerFactory.<init>(OgnlSecurityManagerFactory.java:52)
Call path from entry point to ognl.security.OgnlSecurityManagerFactory.<init>(): 
	at ognl.security.OgnlSecurityManagerFactory.<init>(OgnlSecurityManagerFactory.java:44)
	at ognl.security.OgnlSecurityManagerFactory.getOgnlSecurityManager(OgnlSecurityManagerFactory.java:30)
	at ognl.OgnlRuntime.invokeMethodInsideSandbox(OgnlRuntime.java:1243)
	at ognl.OgnlRuntime.invokeMethod(OgnlRuntime.java:1211)
	at ognl.OgnlRuntime.callAppropriateMethod(OgnlRuntime.java:1920)
	at ognl.ObjectMethodAccessor.callMethod(ObjectMethodAccessor.java:68)
	at ognl.OgnlRuntime.callMethod(OgnlRuntime.java:1996)
	at ognl.ASTMethod.getValueBody(ASTMethod.java:91)
	at ognl.SimpleNode.evaluateGetValueBody(SimpleNode.java:212)
	at ognl.SimpleNode.getValue(SimpleNode.java:236)
	at ognl.Ognl.getValue(Ognl.java:537)
	at ognl.Ognl.getValue(Ognl.java:501)
	at org.thymeleaf.standard.expression.OGNLVariableExpressionEvaluator.executeExpression(OGNLVariableExpressionEvaluator.java:328)
	at org.thymeleaf.standard.expression.OGNLVariableExpressionEvaluator.evaluate(OGNLVariableExpressionEvaluator.java:170)
	at org.thymeleaf.standard.expression.OGNLVariableExpressionEvaluator.evaluate(OGNLVariableExpressionEvaluator.java:95)
	at org.thymeleaf.standard.expression.VariableExpression.executeVariableExpression(VariableExpression.java:166)
	at org.thymeleaf.standard.expression.SimpleExpression.executeSimple(SimpleExpression.java:66)
	at org.thymeleaf.standard.expression.Expression.execute(Expression.java:109)
	at org.thymeleaf.standard.expression.Expression.execute(Expression.java:138)
	at org.thymeleaf.standard.processor.StandardDefaultAttributesTagProcessor.processDefaultAttribute(StandardDefaultAttributesTagProcessor.java:165)
	at org.thymeleaf.standard.processor.StandardDefaultAttributesTagProcessor.process(StandardDefaultAttributesTagProcessor.java:98)
	at org.thymeleaf.engine.ProcessorTemplateHandler.handleStandaloneElement(ProcessorTemplateHandler.java:918)
	at org.thymeleaf.engine.StandaloneElementTag.beHandled(StandaloneElementTag.java:228)
	at org.thymeleaf.engine.TemplateModel.process(TemplateModel.java:136)
	at org.thymeleaf.engine.TemplateManager.process(TemplateManager.java:519)
	at org.thymeleaf.util.LazyProcessingCharSequence.resolveText(LazyProcessingCharSequence.java:78)
	at org.thymeleaf.util.AbstractLazyCharSequence.getText(AbstractLazyCharSequence.java:68)
	at org.thymeleaf.util.AbstractLazyCharSequence.hashCode(AbstractLazyCharSequence.java:132)
	at java.util.HashMap.hash(HashMap.java:339)
	at java.util.HashMap.get(HashMap.java:557)
	at com.oracle.svm.core.jdk.Target_java_nio_charset_Charset.lookup(CharsetSubstitutions.java:74)
	at java.nio.charset.Charset.isSupported(Charset.java:505)
	at com.oracle.svm.jni.JNIJavaCallWrappers.jniInvoke_VA_LIST:Ljava_nio_charset_Charset_2_0002eisSupported_00028Ljava_lang_String_2_00029Z(generated:0)

	at com.oracle.graal.pointsto.constraints.UnsupportedFeatures.report(UnsupportedFeatures.java:126)
	at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:770)
	... 7 more
Caused by: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method java.lang.ClassLoader.defineClass(String, byte[], int, int, ProtectionDomain) is reachable
To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
	at com.oracle.svm.hosted.substitute.AnnotationSubstitutionProcessor.lookup(AnnotationSubstitutionProcessor.java:187)
	at com.oracle.graal.pointsto.infrastructure.SubstitutionProcessor$ChainedSubstitutionProcessor.lookup(SubstitutionProcessor.java:128)
	at com.oracle.graal.pointsto.infrastructure.SubstitutionProcessor$ChainedSubstitutionProcessor.lookup(SubstitutionProcessor.java:128)
	at com.oracle.graal.pointsto.meta.AnalysisUniverse.lookupAllowUnresolved(AnalysisUniverse.java:409)
	at com.oracle.graal.pointsto.infrastructure.WrappedConstantPool.lookupMethod(WrappedConstantPool.java:125)
	at org.graalvm.compiler.java.BytecodeParser.lookupMethodInPool(BytecodeParser.java:4345)
	at com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.lookupMethodInPool(SharedGraphBuilderPhase.java:119)
	at org.graalvm.compiler.java.BytecodeParser.lookupMethod(BytecodeParser.java:4339)
	at org.graalvm.compiler.java.BytecodeParser.genInvokeVirtual(BytecodeParser.java:1700)
	at org.graalvm.compiler.java.BytecodeParser.processBytecode(BytecodeParser.java:5404)
	at org.graalvm.compiler.java.BytecodeParser.iterateBytecodesForBlock(BytecodeParser.java:3436)
	at org.graalvm.compiler.java.BytecodeParser.processBlock(BytecodeParser.java:3243)
	at org.graalvm.compiler.java.BytecodeParser.build(BytecodeParser.java:1109)
	at org.graalvm.compiler.java.BytecodeParser.buildRootMethod(BytecodeParser.java:1003)
	at org.graalvm.compiler.java.GraphBuilderPhase$Instance.run(GraphBuilderPhase.java:84)
	at com.oracle.svm.hosted.phases.SharedGraphBuilderPhase.run(SharedGraphBuilderPhase.java:76)
	at org.graalvm.compiler.phases.Phase.run(Phase.java:49)
	at org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:212)
	at org.graalvm.compiler.phases.Phase.apply(Phase.java:42)
	at org.graalvm.compiler.phases.Phase.apply(Phase.java:38)
	at com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.parse(MethodTypeFlowBuilder.java:223)
	at com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.apply(MethodTypeFlowBuilder.java:357)
	at com.oracle.graal.pointsto.flow.MethodTypeFlow.doParse(MethodTypeFlow.java:313)
	at com.oracle.graal.pointsto.flow.MethodTypeFlow.ensureParsed(MethodTypeFlow.java:302)
	at com.oracle.graal.pointsto.flow.MethodTypeFlow.addContext(MethodTypeFlow.java:103)
	at com.oracle.graal.pointsto.DefaultAnalysisPolicy$DefaultSpecialInvokeTypeFlow.onObservedUpdate(DefaultAnalysisPolicy.java:368)
	at com.oracle.graal.pointsto.flow.TypeFlow.notifyObservers(TypeFlow.java:470)
	at com.oracle.graal.pointsto.flow.TypeFlow.update(TypeFlow.java:542)
	at com.oracle.graal.pointsto.flow.SourceTypeFlowBase.update(SourceTypeFlowBase.java:144)
	at com.oracle.graal.pointsto.BigBang$2.run(BigBang.java:547)
	at com.oracle.graal.pointsto.util.CompletionExecutor.lambda$execute$0(CompletionExecutor.java:173)
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1402)
	... 4 more
Error: Image build request failed with exit status 1
com.oracle.svm.driver.NativeImage$NativeImageError: Image build request failed with exit status 1
	at com.oracle.svm.driver.NativeImage.showError(NativeImage.java:1676)
	at com.oracle.svm.driver.NativeImage.build(NativeImage.java:1426)
	at com.oracle.svm.driver.NativeImage.performBuild(NativeImage.java:1387)
	at com.oracle.svm.driver.NativeImage.main(NativeImage.java:1374)

```

**Comment by oubidar-Abderrahim** [comment]
Issue tracked internally on GR-28790

**Comment by cstancu** [comment]
Fixed by 915e6d2e55b86a88e121e7498a7acc8c4ecd65ee. Note that since we cannot support calling `.ClassLoader.defineClass()` at run time the commit replaces the `@Delete` with a `throw VMError.unsupportedFeature()`. This allows building the image and will just report the issue if that execution path is ever executed.
